arangod/IResearch/IResearchLink.cpp

```diff
@@ -359,6 +358,7 @@ Result IResearchLink::init(velocypack::Slice definition,
             "error finding view for link '" + std::to_string(_id.id()) + "'"};
   }
   TRI_ASSERT(_meta._sortCompression);
+  auto failedInit = scopeGuard([&]() noexcept { _viewGuid = {}; });
```

Comment:
can't we explicitly call `deleteDataStore()` here?

Reply:
I don't want try catch